Documents/public-api.md

```diff
@@ -0,0 +1,34 @@
+# HMDA Platform Public API
+
+This documenatation describes de public HMDA Platform HTTP API
+
+## Institutions
+
+### Search
+
+* `/institutions?domain=<domain>`
+
+   * `GET` - Returns a list of institutions filtered by their email domain. If none are found, an HTTP 404 error code (not found) is returned
```

Comment:
Exactly. So instead of the normal 404 message `Not Found` maybe we could make it a bit more clear. Maybe `Domain Not Found` would do?

The only reason being is that I'm assuming if someone misspells `institutions`, eg `insttttitutions`, we are returning the `Not Found`.

Can we do that? Have a different message for different types of 404s?

If that's too much for now we can always create a new issue later.